### ISSUE TITLE
"gallery_image :responsive, photo, wrap: true" raises "wrong number of arguments (3 for 2)"

### ISSUE THREAD

**Issue by microweb10** [kristianmandrup/rails-gallery#2]
When using the responsive gallery via:

``` haml
.gallery
  = render partial: 'gallery/template/responsive'
  = render partial: 'gallery/responsive', locals: { photos: @property.photos }
```

it loads the file "app/views/gallery/_responsive.html.haml" with contains in the last line:

``` haml
= gallery_image :responsive, photo, wrap: true
```

But in the file "lib/rails-gallery/view_helper.rb" the definition of gallery_image only has 2 arguments:

``` ruby
def gallery_image type, photo
      meth_name = "#{type}_gallery_image"
      validate_gallery_photo! photo
      unless respond_to? meth_name
        raise ArgumentError, "Gallery #{type} is not yet supported. Please add a View helper module for this gallery using the convention followed by the other galleries..." 
      end
      send(meth_name, photo)
    end
```

If I try to remove the last argument (wrap: true) the images are not wrapped.

How can I fix that? 

Because I see in this page "http://tympanus.net/Tutorials/ResponsiveImageGallery/" the gallery works fine.

Thanks

**Comment by kristianmandrup** [comment]
Looks like an error (that was never tested and caught). I recommend you make a clone, then add test cases (specs) for these scenarios... that make a pull request (or just hack the fix directly)

Should be fixed to something like:

``` ruby
    def gallery_image type, photo, options = {}
      meth_name = "#{type}_gallery_image"
      validate_gallery_photo! photo
      unless respond_to? meth_name
        raise ArgumentError, "Gallery #{type} is not yet supported. Please add a View helper module for this gallery using the convention followed by the other galleries..." 
      end
      send meth_name, photo, options
    end

    def gallery_imageset type, imageset, options = {}
      meth_name = "#{type}_gallery_imageset"
      # validate_gallery_imageset! imageset
      unless respond_to? meth_name
        raise ArgumentError, "Gallery #{type} is not yet supported for imageset. Please add a View helper module for this gallery using the convention followed by the other galleries..." 
      end
      send meth_name, imageset, options
    end
```

I'm not sure why the `validate_gallery_imageset!` was not implemented. I guess an imageset is just an array of photos, and so, to validate it simply means iterating through the photos and calling `validate_gallery_photo!` for each one.

Haven't been using this lib for a looong time...

Thanks!

**Comment by microweb10** [comment]
Fixed!!

Once you have the new release you can close this issue.

:)

**Comment by kristianmandrup** [comment]
Released 0.3.3 which should fix this :)
